### PR TITLE
Remove deprecated OpenVPN control plane connectivity

### PR DESCRIPTION
**Pull request by embik**
**What this PR does / why we need it**:
This PR drops support for OpenVPN as solution to connect control plane and worker nodes over network boundaries. Only Konnectivity will be supported from this on. The API field in the `Cluster` spec is a no-op and will no longer be honoured. 

Essentially, this PR rips out all the various if conditions where we were checking if Konnectivity was enabled or not. And that's _a lot_ of places. I would please ask reviewers to go over the actual code changes, this PR looks much scarier because all the fixtures were built with OpenVPN before. Now they are switching over to Konnectivity, so a lot of changes happen in them.

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind cleanup
/kind api-change

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
ACTION REQUIRED: Remove OpenVPN as means to connect control planes and cluster nodes. Existing setups must be migrated to Konnectivity before upgrading
```

**Documentation**:

```documentation
TBD
```

### CODE REVIEW

**Comment by embik** [comment]
/cc @cnvergence @xrstf 

ready for review now.

File: pkg/apis/kubermatic/v1/cluster.go [hunk 1]
```diff
@@ -1459,13 +1458,11 @@ const (
 
 // ExtendedClusterHealth stores health information of a cluster.
 type ExtendedClusterHealth struct {
-	Apiserver         HealthStatus `json:"apiserver,omitempty"`
-	Scheduler         HealthStatus `json:"scheduler,omitempty"`
-	Controller        HealthStatus `json:"controller,omitempty"`
-	MachineController HealthStatus `json:"machineController,omitempty"`
-	Etcd              HealthStatus `json:"etcd,omitempty"`
-	//  Deprecated: OpenVPN will be removed entirely in the future.
-	OpenVPN                      HealthStatus  `json:"openvpn,omitempty"`
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/cluster.go:1468]: I'm not sure if this is breaking stuff and if we should postpone removing fields from the API until 2.27, when we're certain all components in a KKP system are on 2.26 already. Maybe it's overly cautious. Your call.

Comment (embik) [reply pkg/apis/kubermatic/v1/cluster.go:1468]: Fair point. I did follow that policy for the `Cluster` spec field, so I should here as well.

WDYT, should the health status be always up or always down?

**Comment by embik** [comment]
Okay, actually after talking a while to @xrstf about this:

I'll go ahead and close this for now. We should use 2.26 to enforce that everything is on Konnectivity, and 2.27 to remove all the code. Same as we do with Kubernetes versions. I will close this PR and open a more simple one.

**Pull request closed without merging**